seminars/seminar_06/chess.py

```diff
@@ -0,0 +1,96 @@
+# Добавьте в пакет, созданный на семинаре шахматный модуль.
+# Внутри него напишите код, решающий задачу о 8 ферзях.
+# Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они не били друг друга.
+# Вам дана расстановка 8 ферзей на доске, определите, есть ли среди них пара бьющих друг друга.
+# Программа получает на вход восемь пар чисел, каждое число от 1 до 8 - координаты 8 ферзей.
+# Если ферзи не бьют друг друга верните истину, а если бьют - ложь.
+# * - вывести все успешные расстановки
+
+from itertools import combinations
+from random import randint
+
+LENGTH_OF_DESK = 8
+NUM_OF_QUEENS = 8
+NUM_OF_COMB = 4
+
+
+def queen_cuts(queens_coord: list | tuple | set) -> bool:
+    '''
+    Проверяет расстановку 8 ферзей на шахматной доске. Возвращает True, если ферзи не бьют друг друга и False, если хотя бы один бьёт другого
+    :param queens_coord: list(list(2)
+    :return: bool
+    '''
+    # print(queens_coord)
+    for i in range(0, len(queens_coord)):
+        for j in range(i + 1, len(queens_coord)):
+            # print(queens_coord[i], queens_coord[j])
+            if queens_coord[i][0] == queens_coord[j][0] or queens_coord[i][1] == queens_coord[j][1] or abs(
+                    queens_coord[i][0] - queens_coord[j][0]) == abs(queens_coord[i][1] - queens_coord[j][1]):
+                return False
+    return True
+
+
+def get_random_desk_combination(length_of_desk: int) -> list:
+    '''
+    генерирует случайную расстановку ферзей
+    :param length_of_desk: int
+    :return: list
+    '''
+    random_desk_combination = set()
+    while len(random_desk_combination) <= length_of_desk:
+        coord_a, coord_b = randint(1, length_of_desk), randint(1, length_of_desk)
+        random_desk_combination.add((coord_a, coord_b))
+    return list(random_desk_combination)
```

Comment:
Считается, что так быстрее
```suggestion
    return [*random_desk_combination]
```

Reply:
Хотя если вы переводите в в список, почему изначально не сделать списком?